db/mysql/mysqlClient.go

```diff
@@ -0,0 +1,140 @@
+package mysql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/go-sql-driver/mysql"
+	guuid "github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/spf13/viper"
+
+	"github.com/abdulmoeid7112/student-backend-portal/config"
+	"github.com/abdulmoeid7112/student-backend-portal/db"
+	"github.com/abdulmoeid7112/student-backend-portal/models"
+)
+
+//ErrorFooUserNotFoundError.
+var ErrorFooUserNotFoundError = errors.New("User not found")
+
+//ErrorFooIDEmpty.
+var ErrorFooIDEmpty = errors.New("id is not empty")
+
+//ErrorFooFailedError.
+var ErrorFooFailedError = errors.New("failed to connect")
+
+//ErrorFooFailedAdd.
+var ErrorFooFailedAdd = errors.New("failed to add")
+
+//ErrorFooFailedUpdate.
+var ErrorFooFailedUpdate = errors.New("failed to update")
+
+//ErrorFooFailedDelete.
+var ErrorFooFailedDelete = errors.New("failed to delete")
+
+//ErrorFooFailedRetrieve.
+var ErrorFooFailedRetrieve = errors.New("failed to retrieve")
+
+type mysqlStd struct {
+	db *sqlx.DB
+}
+
+func (m mysqlStd) AddStudent(student *models.Student) (string, error) {
+	if student.ID != "" {
+		return "", ErrorFooIDEmpty
+	}
+	student.ID = guuid.New().String()
+	names := student.Names()
+	if _, err := m.db.NamedExec(fmt.Sprintf(`INSERT INTO Student (%s) VALUES(%s)`, strings.Join(names, ","), strings.Join(mkPlaceHolder(names, ":", func(name, prefix string) string {
+		return prefix + name
+	}), ",")),
+		student); err != nil {
+		return "", ErrorFooFailedAdd
+	}
+	return "", nil
+}
+
+func (m mysqlStd) UpdateStudent(student *models.Student) error {
+	names := student.Names()
+	if _, err := m.db.NamedExec(fmt.Sprintf(`UPDATE Student SET %s`, strings.Join(mkPlaceHolder(names, "=:", func(name, prefix string) string {
+		return name + prefix + name
+	}), ",")), student); err != nil {
+		return ErrorFooFailedUpdate
+	}
+	return nil
+}
+
+func (m mysqlStd) GetStudentByID(id string) (*models.Student, error) {
+	var std models.Student
+	if err := m.db.Get(&std, fmt.Sprintf(`SELECT * FROM Student WHERE id='%s'`, id)); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ErrorFooUserNotFoundError
+		}
+		return nil, err
+	}
+	fmt.Printf("%v\n", std.Map())
+	return &std, nil
+}
+
+func (m mysqlStd) ListStudent() ([]*models.Student, error) {
+	var stdList []*models.Student
+	var std models.Student
+	rows, err := m.db.Query(fmt.Sprintf(`SELECT * FROM Student`))
```

Comment:
Need to add where clause... based on `filter map[string]interface{}` 
Where filter will hold key value pair based on the student model.

Also need to add limit and offset for proper pagination support.

Signature would look like
```
func (m mysqlStd) ListStudent() ([]*models.Student, error) {
```
example usage
```
students,err client.ListStudent(map[string]interface{"age":20"},10, 20)
```